app/models/git_repository.rb

```diff
@@ -114,6 +114,27 @@ def file_changed?(sha1, sha2, file)
     executor.execute!("cd #{pwd}", "git diff --quiet --name-only #{sha1}..#{sha2} #{file}")
   end
 
+  def directory_contents(git_ref, path: '.')
+    cmd = "git ls-tree -r #{git_ref} --name-only --full-tree #{path}"
+    valid, output = run_single_command(cmd) { |line|
+      if block_given?
+        yield line.chomp
+      else
+        line.chomp
+      end
+    }
+    Rails.logger.error("Could not list the files in directory '#{path}'. Git Ref '#{git_ref}'") unless valid
+    output
+  end
+
+  def file_contents(git_ref, file_path)
+    #first command disables less as default pager
+    cmds = ["git config core.pager ''", "git show -q #{git_ref}:#{file_path}"]
+    valid, output = run_commands(cmds) { |line| line }
+    Rails.logger.error("Could not retrieve file contents for '#{file_path}'. Git Ref '#{git_ref}'") unless valid
+    output.join
+  end
```

Comment:
Yeah... I was wondering about which way to go, now that we're listing each Role config file and fetching it's contents. I'm conscious there's not going to be a lot of Roles, but I was discussing this with @henders as well, and we were wondering if it's not actually faster to use the TerminalExecutor. Fetching the contents of each file will lead to a separate API call for each file. Using the TerminalExecutor, we already have all the files in the local cached repo, so, it should be faster than using the API.

I should also mention that, due to some new ideas we've been thinking about, which I'll explain by email, there's a good chance the User will be triggering this logic when creating a new Release, or creating a new Role. Either way,  in terms of client-side user experience, we'd like to make this calls as soon as possible.